modules/core/src/va_intel.cpp

```diff
@@ -85,7 +85,10 @@ Context& initializeContextFromVA(VADisplay display, bool tryInterop)
 #else  // !HAVE_VA
 
 #   ifdef HAVE_VA_INTEL
-    if (tryInterop)
+
+    Context& va_ctx = OpenCLExecutionContext::getCurrent().getContext();
+    VAAPIInterop* interop = va_ctx.getUserContext<VAAPIInterop>().get();
+    if (tryInterop && interop == NULL)
```

Comment:
I think I understand. I am on it

Reply:
But how to handle VA (non-Intel)? Is it correct to return a context without interop?

Reply:
Also. what happens if it is called multiple times and the context is already with VADisplay? I am asking specifically because ```initializeContextFromVA()``` is called from withing VideoWriter and VideoCapture.

Reply:
I myself have a proposal for what context handling should maybe look like based on my  [4.x fork](https://github.com/kallaballa/opencv/tree/GCV): https://github.com/kallaballa/GCV/blob/main/src/tetra/tetra-demo.cpp

It basically creates a context for each interop-ability (OpenGL/VAAPI) copies them and when they are needed it binds them. So only one ```initializeContextFromVA()```